### PR TITLE
[CC-18] Set up `@cc/ui-chakra` and `@cc/ui-tailwind`

### PR DESCRIPTION
**Pull request by NorrisWu0**
### New Feature
- [x] Set up `@cc/ui-chakra`, a react UI library with `@chakra-ui` and `@storybook`.
- [x] Set up `@cc/ui-tailwind`, a react UI library with `tailwindcss`.
- [x] Install `@cc/ui-chakra` onto `galaxy` project.
- [x] Install `@cc/ui-tailwind` onto `startrek` project.

### Chore
- [x] Bump version of `@types/react` in `@cc/ui-chakra` to fix issue with type mismatch on `React.ReactNode`.
- [x] Amend eslint rule `no-console` to warning level.'

### Notes

Additional comments are made outside this pull request, see the list below.

1. Questionable placeholder URL used in button onClick events. @kritomomo 
This change has been reflected in [fix: incorrect placeholder url](https://github.com/courtcanva/cc-multiverse/pull/3/commits/052b97e5e46b0fe28f7da14225bcbde6d0bc0bb3)

2. Lack of clarity around the usage of @cc/ui-chakra and @cc/ui-tailwind
Fixed in commit [docs: clarify purpose of ui libraries](https://github.com/courtcanva/cc-multiverse/pull/3/commits/43d2b22373c78316aeaa24050a7977f26c7bc0f4) @kritomomo 

3. Use of [interim type definition - VFC](https://basarat.gitbook.io/typescript/tsx/react#void-function-components) in components @kritomomo 
Updated as per comment in [refactor: remove storybook from @cc/ui-tailwind](https://github.com/courtcanva/cc-multiverse/pull/3/commits/ef2c802f9297e8f134a82fda2d48b983d608e3eb)

### Known issue
1. `storybook` currently does not work with `@cc/ui-tailwind`, however, the components themselves work as expected in `startrek` project.

### CODE REVIEW

File: apps/galaxy/src/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Stack, Button } from "@cc/ui-chakra";
+
+const NEVER_GONNA_GIVE_YOU_UP = "https://www.youtube.com/watch?v=dQw4w9WgXcQ";
```
Comment (tatoMa) [inline apps/galaxy/src/pages/index.tsx:3]: Serious?

Comment (NorrisWu0) [reply apps/galaxy/src/pages/index.tsx:3]: Idea brought from @kritomomo 😂

Comment (tatoMa) [reply apps/galaxy/src/pages/index.tsx:3]: So how can I approve this pr?

Comment (NorrisWu0) [reply apps/galaxy/src/pages/index.tsx:3]: Almost home, will fix this asap. 👌

**Comment by NorrisWu0** [comment]
Okay, since this PR is opened, there have been these changes made to these comments...

> Questionable placeholder URL used in button onClick events.
This change has been reflected in [fix: incorrect placeholder url](https://github.com/courtcanva/cc-multiverse/pull/3/commits/052b97e5e46b0fe28f7da14225bcbde6d0bc0bb3)

> Lack of clarity around the usage of `@cc/ui-chakra` and `@cc/ui-tailwind`
Fixed in commit [docs: clarify purpose of ui libraries](https://github.com/courtcanva/cc-multiverse/pull/3/commits/43d2b22373c78316aeaa24050a7977f26c7bc0f4)

> Use of [interim type definition - VFC](https://basarat.gitbook.io/typescript/tsx/react#void-function-components) in components
Updated as per comment in [refactor: remove storybook from @cc/ui-tailwind](https://github.com/courtcanva/cc-multiverse/pull/3/commits/ef2c802f9297e8f134a82fda2d48b983d608e3eb)

But yea it should be good to be reviewed again now. 👍 @tatoMa

**Comment by chui-mel** [comment]
> Okay, since this PR is opened, there have been these changes made to these comments...
> 
> > Questionable placeholder URL used in button onClick events.
> 
> This change has been reflected in [fix: incorrect placeholder url](https://github.com/courtcanva/cc-multiverse/pull/3/commits/052b97e5e46b0fe28f7da14225bcbde6d0bc0bb3)
> 
> > Lack of clarity around the usage of `@cc/ui-chakra` and `@cc/ui-tailwind`
> 
> Fixed in commit [docs: clarify purpose of ui libraries](https://github.com/courtcanva/cc-multiverse/pull/3/commits/43d2b22373c78316aeaa24050a7977f26c7bc0f4)
> 
> > Use of [interim type definition - VFC](https://basarat.gitbook.io/typescript/tsx/react#void-function-components) in components
> 
> Updated as per comment in [refactor: remove storybook from @cc/ui-tailwind](https://github.com/courtcanva/cc-multiverse/pull/3/commits/ef2c802f9297e8f134a82fda2d48b983d608e3eb)
> 
> But yea it should be good to be reviewed again now. 👍 @tatoMa

can you move this part of information to PR description?

File: packages/ui-tailwind/src/display/Icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+import NotFoundIcon from "@/assets/icons/404.svg";
```
Comment (kritomomo) [inline packages/ui-tailwind/src/display/Icon/Icon.tsx:3]: It looks like you haven't used this icon inside the component

File: packages/ui-tailwind/src/display/Icon/Icon.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      >
+        <path
+          d="M40 0C17.9032 0 0 17.9032 0 40C0 62.0968 17.9032 80 40 80C62.0968 80 80 62.0968 80 40C80 17.9032 62.0968 0 40 0ZM40 72.2581C22.2097 72.2581 7.74194 57.7903 7.74194 40C7.74194 22.2097 22.2097 7.74194 40 7.74194C57.7903 7.74194 72.2581 22.2097 72.2581 40C72.2581 57.7903 57.7903 72.2581 40 72.2581ZM27.0968 37.4194C29.9516 37.4194 32.2581 35.1129 32.2581 32.2581C32.2581 29.4032 29.9516 27.0968 27.0968 27.0968C24.2419 27.0968 21.9355 29.4032 21.9355 32.2581C21.9355 35.1129 24.2419 37.4194 27.0968 37.4194ZM52.9032 27.0968C50.0484 27.0968 47.7419 29.4032 47.7419 32.2581C47.7419 35.1129 50.0484 37.4194 52.9032 37.4194C55.7581 37.4194 58.0645 35.1129 58.0645 32.2581C58.0645 29.4032 55.7581 27.0968 52.9032 27.0968ZM40 47.7419C33.5161 47.7419 27.4194 50.5968 23.2581 55.5806C21.8871 57.2258 22.1129 59.6613 23.7581 61.0323C25.4032 62.3871 27.8387 62.1774 29.2097 60.5323C31.8871 57.3226 35.8226 55.4677 40 55.4677C44.1774 55.4677 48.1129 57.3064 50.7903 60.5323C52.0968 62.0968 54.5161 62.4516 56.2419 61.0323C57.8871 59.6613 58.0968 57.2258 56.7419 55.5806C52.5806 50.5968 46.4839 47.7419 40 47.7419Z"
+          fillOpacity="0.8"
+        />
+      </svg>
```
Comment (kritomomo) [inline packages/ui-tailwind/src/display/Icon/Icon.tsx:24]: Can you replace this part with 404.svg, and in the next ticket, you may need to configure this https://github.com/courtcanva/cc-app/commit/32195a1c8d4dffd395f60ad670cc8dd5912ca4ad

**Comment by kritomomo** [comment]
Regarding the classification of components in UI Library, at least at the code level, I think Button should not be classified in Form. I took a look at chakra's repo, which puts all components in the `components` folder. [chakra repo](https://github.com/chakra-ui/chakra-ui/tree/main/packages/components)
image
For the material ui, they put the button into the base folder
image
Maybe this part need @tatoMa ‘s suggestion

**Review by kritomomo — APPROVED** [review]

**Comment by tatoMa** [comment]
> Regarding the classification of components in UI Library, at least at the code level, I think Button should not be classified in Form. I took a look at chakra's repo, which puts all components in the `components` folder. [chakra repo](https://github.com/chakra-ui/chakra-ui/tree/main/packages/components) image For the material ui, they put the button into the base folder image Maybe this part need @tatoMa ‘s suggestion

Totally agree with you. My opinion is button components can be inside of the base components. But it is not set in stone though.

File: apps/startrek/src/styles/globals.css [hunk 4]
```diff
@@ -0,0 +1,17 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+@layer base {
+  html {
+    font-family: Lexend, sans-serif;
```
Comment (tatoMa) [inline apps/startrek/src/styles/globals.css:7]: Are we using this font or it is a placeholder?

File: packages/ui-chakra/src/form/Button/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    error: {
+      bg: "#6F4E93",
+    },
+  };
+
+  return (
+    <CKButton onClick={onClick} color="#FFFFFF" {...buttonVariants[variant]}>
+      {children}
+    </CKButton>
```
Comment (tatoMa) [inline packages/ui-chakra/src/form/Button/Button.tsx:33]: Do you think we need useRef in the Button component or just leave it in the future if needed?

**Review by tatoMa — APPROVED** [review]

File: packages/ui-tailwind/src/display/Icon/Icon.tsx (hunk 2 above)
Comment (NorrisWu0) [reply packages/ui-tailwind/src/display/Icon/Icon.tsx:3]: Okay... That's a mistake here, I think I will keep this here for now, I should be configuring webpack to load svg in the next pr (assuming I'm understanding this correctly)

File: packages/ui-tailwind/src/display/Icon/Icon.tsx (hunk 3 above)
Comment (NorrisWu0) [reply packages/ui-tailwind/src/display/Icon/Icon.tsx:24]: Kool kool, will make sure this is configured in next pr.

File: apps/startrek/src/styles/globals.css (hunk 4 above)
Comment (NorrisWu0) [reply apps/startrek/src/styles/globals.css:7]: This font was not referenced from our UI, it's just a placeholder font to check if the configuration is correct in tailwindcss

File: packages/ui-chakra/src/form/Button/Button.tsx (hunk 5 above)
Comment (NorrisWu0) [reply packages/ui-chakra/src/form/Button/Button.tsx:33]: Hmm, I haven't had the experience to use ref for buttons yet, but yea, we can add them when we need to I think.

**Pull request merged**